doc/memos/rdm0002.md

```diff
@@ -0,0 +1,2226 @@
+implementation SHOULD NOT implement reentrancy, it MUST handle concurrent calls
+between the IRQ handler and API functions.
+
+The BH processor can be implemented dependent or independent of the network
+stack. A network stack independent solution is preferred in order to reuse
+functionality between different network stacks.
+
+The term "Bottom Half" was originally introduced by the Linux kernel. See
+[Top and Bottom Halves](http://www.makelinux.net/ldd3/chp-10-sect-4.shtml)
+
+## 4.3 Radio HAL
+
+The Radio HAL is defined by the Radio HAL API which consists of three main
+components: Radio Operations, Event Notification, and the Device Specific
+IEEE802.15.4 HAL implementation.
+
+The Radio HAL Implementation provides a set of functionalities to control the
+operation of the device, to process the IRQ handler and to receive event
+notifications from the device.
+
+### 4.3.1 Radio Operations
+The Radio Operations (`radio_ops`) interface exposes operations that are common
+to control 802.15.4 devices and to request their hardware capabilities
+information (i.e., MAC acceleration hardware)
+
+The interface defines a collection of mandatory functions:
+- Set the transceiver state
+- Set the PHY configuration (channel, tx power, etc)
+- Load and transmit a frame
+- Get device capabilities
+
+The interface provides a collection of optional functions that may or may not
+be implemented, dependent on the hardware acceleration features of a device.
+These functions include:
+- Read the number of retransmission attempts.
+- Set address filter addresses (extended, short, PAN ID).
+- Set CSMA-CA backoff parameters.
+
+All `radio_ops` functions are non-blocking and some of them follow a
+Request/Confirm scheme which means, the end of a request is indicated by a
+confirmation function. The confirmation function may be polled. In such case,
+the function uses standard error codes to indicate the status (success, error
+or request has not finished).
+
+The full list of functions can be found in the Interface Definition section.
+
+### 4.3.2 Event Notification
+
+The Radio HAL provides an Event Notification mechanism to inform the upper
+layer about an event (a packet was received, a transmission finished, etc).
+
+The upper layer can subscribe to these events to perform different actions. As
+an example, a MAC layer would subscribe to the RX done event to allocate the
+received packet. The TX done event is commonly used to release resources or
+update statistics.
+
+As described before, the Event Notification mechanism can be called during ISR
+or thread context (BH processor). Thus, this must be taken into consideration
+for the implementation of the Event Notification callback (e.g the `RX_DONE`
+event might post an event to an event queue in order to fetch the packet or
+`PLL_LOCK` might unlock a mutex).
+
+The full list of events and implications are defined in the Interface
+Definition section.
```

Comment:
`PLL_LOCK` is not mentioned in the Interface Definition section at all. Is that intentional?

Reply:
Kind of. The RDM does not prevent adding such an event, but we didn't include it in the final implementation because it's kind of a niche use case. We can still leave it there to point out that other events can be introduced if needed.

Reply:
> We can still leave it there to point out that other events can be introduced if needed.

Maybe rather remove the example here and add a mention of the introduction of other events still?

Reply:
done!